actions/retrieve-bulk-chain-events.js

```diff
@@ -0,0 +1,76 @@
+import "dotenv/config";
+import db from "../db/index.js";
+import events from "../modules/retrieve-bulk-actions/index.js";
+import DAOService from "../services/dao-service.js";
+import { error, info } from "../utils/logger-handler.js";
+export const name = "retrieve-bulk-chain-events";
+export const schedule = "30 * * * * *";
+export const description = "retrieving bulk chain events";
+export const author = "clarkjoao";
+
+async function retriveEvents(network, contract, fromBlock, toBlock) {
+  const arrEvents = Object.keys(events).map(async (action) => {
+    info(
+      `Retrieving ${action} events from block ${fromBlock} to ${toBlock} at ${network.name}`
+    );
+
+    const eventsBlock = await contract.network[action]({
+      fromBlock,
+      toBlock,
+    });
+
+    if (eventsBlock.length)
+      return await events[action](eventsBlock, network, contract);
+
+    return [];
+  });
+
+  Promise.all(arrEvents).then((res) => {
+    console.log("done", res);
+  });
+}
+
+export async function action() {
+  info("Starting retrieving bulk chain events");
```

Comment:
the Idea is for us to have one action per thing, not an action that is the bulk of all actions ^^'

Reply:
OK, so i will rewrite